src/big/big_float.cr

```diff
@@ -221,3 +225,10 @@ class String
     BigFloat.new(self)
   end
 end
+
+module Math
+  def frexp(value : BigFloat)
+    frac = LibGMP.mpf_get_d_2exp(out exp, value)
+    {frac, exp}
```

Comment:
Implementation for exact calculation (for example, for exact hashing) should look like
```crystal
module Math
  def frexp(value : BigFloat)
    LibGMP.mpf_get_d_2exp(out exp, value)
    frac = BigFloat.new { |mpf|
      if exp >= 0      
        LibGMP.mpf_div_2exp(mpf, self, exp)
      else
        LibGMP.mpf_mul_2exp(mpf, self, -exp)
      end
    }
    {frac, exp}
  end
end
```